app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectInformationActivity.kt

```diff
@@ -64,33 +67,61 @@ class ProjectInformationActivity : AppCompatActivity() {
     private fun readNextVideo() {
         if (current + 1 < videosUris.size) {
             current++
-            videoView.setVideoURI(videosUris[current])
-            videoView.start()
+            updateVideoState()
         }
     }
 
     private fun readPrevVideo() {
         if (current > 0) {
             current--
-            videoView.setVideoURI(videosUris[current])
-            videoView.start()
+            updateVideoState()
         }
     }
 
+    private fun updateVideoState() {
+        val localUri = videosUris[current]
+        /* onlineUri is used as unique identifier for subtitles */
+        val onlineUri = projectVar.videoURI[current]
+        videoView.setVideoURI(localUri)
+        metadataDB.getSubtitlesFromVideo(
+            onlineUri,
+            Locale.ENGLISH.language,
+            { subs ->
+                subs?.let {
+                    videoView.addSubtitleSource(
```

Comment:
The subtitles are downloaded only when the video is played right ?
Maybe downloading once, when videos are downloaded could be better and could avoid to have no subtitle at the beginning if we have slow internet speed.

Reply:
Yes good idea
It will be in the next commit